src/app/files/mod.rs

```
@@ -142,51 +146,60 @@ impl Safe {
-                        ))
-                    })?;
-
-                Ok((version, files_map))
-            }
-            Err(Error::EmptyContent(_)) => {
-                warn!("FilesContainer found at \"{:?}\" was empty", safe_url);
-                Ok((0, FilesMap::default()))
-            }
-            Err(Error::ContentNotFound(_)) => Err(Error::ContentNotFound(
-                ERROR_MSG_NO_FILES_CONTAINER_FOUND.to_string(),
-            )),
-            Err(Error::VersionNotFound(_)) => Err(Error::VersionNotFound(format!(
-                "Version '{}' is invalid for FilesContainer found at \"{}\"",
-                safe_url.content_version().unwrap_or(0),
-                safe_url,
-            ))),
-            Err(err) => Err(Error::NetDataError(format!(
-                "Failed to get current version: {}",
-                err
-            ))),
+    ) -> Result<(VersionHash, FilesMap)> {
+        // fetch register entries and wrap errors
+        let entries = self
+            .fetch_register_entries(&safe_url)
+            .await
+            .map_err(|e| match e {
+                Error::ContentNotFound(_) => {
+                    Error::ContentNotFound(ERROR_MSG_NO_FILES_CONTAINER_FOUND.to_string())
+                }
+                Error::HashNotFound(_) => Error::VersionNotFound(format!(
+                    "Version '{}' is invalid for FilesContainer found at \"{}\"",
+                    match safe_url.content_version() {
+                        Some(v) => v.to_string(),
+                        None => "None".to_owned(),
+                    },
+                    safe_url
+                )),
+                err => Error::NetDataError(format!("Failed to get current version: {}", err)),
+            })?;
+
+        // take the 1st entry (TODO Multiple entries)
+        if entries.len() > 1 {
+            unimplemented!("Multiple file container entries not managed, this happends when 2 clients write concurrently to a file container");
```

I think it'll be better to return an error so it's properly reported to the user explaining the limitation of this lib/API, otherwise CLI (or the app using the lib) crashes and provides a dump stack trace report which users won't understand.